src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -533,6 +532,12 @@ public class Options
             [Argument(ArgumentType.AtMostOnce, HelpText = "Character to use to escape quotes inside quoted fields. It can't be a character used as separator.", ShortName = "escapechar")]
             public char EscapeChar = Defaults.EscapeChar;
 
+            /// <summary>
+            /// If true, empty float fields will be loaded as NaN. If false, they'll be loaded as 0. Default is false.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "If true, empty float fields will be loaded as NaN. If false, they'll be loaded as 0. Default is false.", ShortName = "imputefloat")]
+            public bool ImputeEmptyFloats = Defaults.ImputeEmptyFloats;
```

Comment:
I'm definitely not happy with the name I chose for this option, and for its doc. I think the name "field" can mean many things (but inside the code of the TextLoader a "field" is the value found on a given column on a given row). Also, this option is supposed to be related to both Single and Doubles, so "float" is misleading. I'm not sure how to tackle this.

Reply:
I am not sure 'Impute' fits the terminology here. A few options to consider: ParseEmptyFloatsAsNan, ParseEmptyRealsAsNan, EmptyFieldsToNan, EmptyValuesToNan, etc.

---
In reply to: [434309670](https://github.com/dotnet/machinelearning/pull/5198#discussion_r434309670) [](ancestors = 434309670)

Reply:
I used the word "impute" as that was the original name  @justinormont suggested here [https://github.com/dotnet/machinelearning/issues/4132#issuecomment-522786766](https://github.com/dotnet/machinelearning/issues/4132#issuecomment-522786766)
I wouldn't know if that's a common term in which our users could understand the feature by reading it.

As for the names you provided, perhaps EmptyRealsAsNan would fit well, as it avoids the word Field, and "Reals" could imply Floats or Doubles.

I will leave unaddressed right now, and see if we get any suggestion on this. It's only a variable name change, and I'll be able to make the change before merging the PR.

---
In reply to: [434886717](https://github.com/dotnet/machinelearning/pull/5198#discussion_r434886717) [](ancestors = 434886717,434309670)

Reply:
I've decided to go for MissingRealsAsNan , since it's not only the Empty fields that get mapped as NaNs, but also the ones that only have whitespace, and now also the ones were there are missing columns in a given row.

---
In reply to: [435163479](https://github.com/dotnet/machinelearning/pull/5198#discussion_r435163479) [](ancestors = 435163479,434886717,434309670)

Reply:
In that case you should fix the documentation above to read "If true, empty or missing float and double fields will be loaded as NaN"

---
In reply to: [437077933](https://github.com/dotnet/machinelearning/pull/5198#discussion_r437077933) [](ancestors = 437077933,435163479,434886717,434309670)

Reply:
Yeah, the documentation reads:
```
            /// <summary>
            /// If true, missing real fields (i.e. double or single fields) will be loaded as NaN.
            /// If false, they'll be loaded as 0. Default is false.
            /// A field is considered "missing" if it's empty, if it only has whitespace, or if there are missing columns
            /// at the end of a given row.
            /// </summary>
```

---
In reply to: [437170658](https://github.com/dotnet/machinelearning/pull/5198#discussion_r437170658) [](ancestors = 437170658,437077933,435163479,434886717,434309670)